compiler/optimizer/VPHandlers.cpp

```diff
@@ -2161,8 +2161,7 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
           && base->isNonNullObject()
           && base->getClassType()
           && base->getClassType()->asFixedClass()
-          && base->getClass()
-          && TR::Compiler->cls.isClassInitialized(vp->comp(), base->getClass()))
```

Comment:
It's explained in my first commit message. And I'll forward you an email from Devin with more details.